components/up.libsonnet

```
@@ -109,6 +104,48 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
   },
 
+  withRead:: {
+    local u = self,
+    config+:: {
+      readEndpoint: error 'must provide read endpoint',
+    },
+    deployment+: {
+      spec+: {
+        template+: {
+        spec+: {
+          containers: [
+              if c.name == 'observatorium-up' then c {
+                args+: ['--endpoint-read=' + up.config.readEndpoint]
+              } else c
+              for c in super.containers
+            ],
+          },
+        },
+      },
+    },
+  },
+
+  withWrite:: {
```

How about following just to be consistent
```suggestion
  withWriteEndpoint:: {
```

---

done